### DISCUSSION TITLE
How to load fonts that are not TTF?

### DISCUSSION THREAD

**Discussion by jimadamskye** [ocornut/imgui#6831] [New Users Build/Link/Run issues ONLY!]
I am writing an app in rust and would like to use a font from a webpage. The font is a woff2 type so is not a ttf or otf. Looking at the code I didn't see a way to get this to work. The code certainly chokes when it attempts to read that kind of font. Are there any options? I see some crates that will parse the WOFF2 font but I don't know how to integrate them into imgui.

**Comment by PathogenDavid** [comment]
Have you tried [imgui_freetype](https://github.com/ocornut/imgui/tree/master/misc/freetype)?

I can't say I've tried it myself but FreeType is supposed to support WOFF so it might just work.

**Comment by jimadamskye** [comment]
I guess we would have to port this to rust since that is where I am….

**Reply by PathogenDavid** [reply]
Assuming you're using the [`imgui`](https://crates.io/crates/imgui/) crate, I believe you can enable FreeType support by enabling the `freetype` feature in your `Cargo.toml`.

**Comment by jimadamskye** [comment]
Where are we supposed to find out about features? I thought I read something about a new improvement that would better document them but I haven’t seen evidence of it. I only find out about some useful features in Reddit comments like this one. Sent from my iPhoneOn Sep 23, 2023, at 10:51 PM, David Maas ***@***.***> wrote:﻿
Assuming you're using the imgui crate, I believe you can enable FreeType support by enabling the freetype feature in your Cargo.toml.

**Reply by PathogenDavid** [reply]
[They're listed on docs.rs](https://docs.rs/crate/imgui/0.11.0/features). (You can get to that page using the button in the top bar labeled feature flags.)

You can also just peek at [the crate's Cargo.toml](https://github.com/imgui-rs/imgui-rs/blob/8b2722d00b5d5a0d8077881f6f569844ca1e9f42/imgui/Cargo.toml#L27).
